chain/vm/fvm.go

```diff
@@ -707,3 +715,28 @@ func (r *xRedirect) MarshalCBOR(w io.Writer) error {
 
 	return nil
 }
+
+func DecodeEvents(input []byte, legacy bool) ([]cbg.CBORMarshaler, error) {
```

Comment:
I'd suggest just passing in the network version here instead of a boolean.

Reply:
Note: you could also just check the codec (which will be 0 for legacy events).

Reply:
Checking the codec should work (and it's simpler).